run_experiment.py

```diff
@@ -122,19 +131,39 @@ def parse_arguments():
     dir_opts.add_argument('-o', '--output_file_name', help='The full path name to use when saving the plot.')
 
     args = parser.parse_args()
+
+    if args.log_dir is None:
+        args.log_dir = args.model_dir + 'losses.log'
+
     return args
 
 
 def main(args):
     """ The primary entry point for carrying out experiments on pretrained models.
     """ 
-    experiment_name = args.experiment
+    formatter = logging.Formatter('%(asctime)s %(levelname)s - %(funcName)s: %(message)s',
+                                  "%H:%M:%S")
+    logger = logging.getLogger(__name__)
+    logger.setLevel(args.log_level.upper())
+    stream = logging.StreamHandler()
+    stream.setLevel(args.log_level.upper())
+    stream.setFormatter(formatter)
+    logger.addHandler(stream)
+
+    experiment_name = args.model_dir
     dataset = read_dataset_from_specs('experiments/{}/specs.json'.format(experiment_name))
 
     if args.visualisation == 'snapshot_recon':
         viz_list = []
         epoch_list = []
-        model_list = load_model(directory='experiments/{}'.format(experiment_name), load_snapshots=True)
+
+        model_list = load_model(
+            directory='experiments/{}'.format(experiment_name),
+            load_snapshots=True
+            )
+        if not model_list:
+            raise Exception('Please target a trained model which has snapshot models stored in the form:\nmodel-0.pt, model-1.pt, etc.')
+
         for epoch_index, model in model_list:
             model.eval()
             viz_list.append(Visualizer(model=model, model_dir='experiments/{}'.format(experiment_name), dataset=dataset, save_images=False))
```

Comment:
general comments the `save_images=` is not from me but the other repo. I would never have done that but rather always return the image and then have a function `save_images()`that saves the return if needed. 
I think it would be cleaner, but that would mean a lot of changes so if you don't want to do it / don't agree it's fine.